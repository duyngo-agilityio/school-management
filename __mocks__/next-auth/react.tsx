import { createContext } from "react";

export const useSession = () => ({
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
});

export type Session = {
  data: {
    user: {
      token: string;
      email: string;
      id: string;
      name: string;
    };
    expires: string;
  };
  status: "loading" | "authenticated" | "unauthenticated";
  update: (data?: Session) => Promise<Session | null>;
};

const INITIAL_SESSION: Session = {
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
};

const SessionContext = createContext(INITIAL_SESSION);

export const SessionProvider = ({ children }: React.PropsWithChildren) => {
  <SessionContext.Provider value={INITIAL_SESSION}>
    {children}
  </SessionContext.Provider>;
};
