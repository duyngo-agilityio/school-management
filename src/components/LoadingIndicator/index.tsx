// Components
import { Box, Spinner } from '@chakra-ui/react';

const LoadingIndicator = () => (
  <Box
    h="100vh"
    display="flex"
    alignItems="center"
    textAlign="center"
    justifyContent="center"
  >
    <Spinner size="lg" />
  </Box>
);

export default LoadingIndicator;
