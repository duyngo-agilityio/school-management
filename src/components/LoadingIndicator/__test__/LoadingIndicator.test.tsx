import { render } from '@testing-library/react';

// Components
import LoadingIndicator from '..';

describe('LoadingIndicator component', () => {
  test('should match snapshot', () => {
    const { container } = render(<LoadingIndicator />);

    expect(container).toMatchSnapshot();
  });
});
