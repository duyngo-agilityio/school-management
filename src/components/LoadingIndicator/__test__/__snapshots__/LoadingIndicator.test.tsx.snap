// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator component should match snapshot 1`] = `
<div>
  <div
    class="css-1t7pcsk"
  >
    <div
      class="chakra-spinner css-1y7joxr"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
